chia/full_node/mempool.py

```diff
@@ -412,13 +415,15 @@ def create_bundle_from_mempool_items(
                     item_cost + cost_sum > self.mempool_info.max_block_clvm_cost
                     or fee + fee_sum > DEFAULT_CONSTANTS.MAX_COIN_AMOUNT
                 ):
-                    break
+                    continue  # Keep looking for smaller items to fill the block
                 coin_spends.extend(unique_coin_spends)
                 additions.extend(unique_additions)
                 sigs.append(item.spend_bundle.aggregated_signature)
                 cost_sum += item_cost
                 fee_sum += fee
                 processed_spend_bundles += 1
+                if cost_sum / self.mempool_info.max_block_clvm_cost >= 0.998:  # 99.8% full seems a reasonable target
```

Comment:
I think a better model is to stop when the cost left to fill in the block is smaller than some absolute threshold. That way it doesn't depend on the fill rate